cms/djangoapps/contentstore/rest_api/v0/views/advanced_settings.py

```diff
@@ -0,0 +1,162 @@
+""" API Views for course advanced settings """
+
+import edx_api_doc_tools as apidocs
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from edx_rest_framework_extensions.auth.session.authentication import SessionAuthenticationAllowInactiveUser
+from opaque_keys.edx.keys import CourseKey
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from xmodule.modulestore.django import modulestore
+
+from cms.djangoapps.models.settings.course_metadata import CourseMetadata
+from common.djangoapps.student.auth import has_studio_read_access, has_studio_write_access
+from openedx.core.lib.api.authentication import BearerAuthenticationAllowInactiveUser
+from openedx.core.lib.api.view_utils import DeveloperErrorViewMixin, verify_course_exists
+from ..serializers import CourseAdvancedSettingsSerializer
+from ....views.course import update_course_advanced_settings
+
+
+class AdvancedCourseSettingsView(DeveloperErrorViewMixin, APIView):
+    """
+    View for getting and setting the advanced settings for a course.
+    """
+
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser,
+    )
+    permission_classes = (IsAuthenticated,)
+
+    @apidocs.schema(
+        parameters=[apidocs.string_parameter("course_id", apidocs.ParameterLocation.PATH, description="Course ID")],
+        responses={
+            200: CourseAdvancedSettingsSerializer,
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        },
+    )
+    @verify_course_exists
+    def get(self, request: Request, course_id: str):
+        """
+        Get an object containing all the advanced settings in a course.
+
+        **Example Request**
+
+            GET /api/contentstore/v0/advanced_settings/{course_id}
+
+        **Response Values**
+
+        If the request is successful, an HTTP 200 "OK" response is returned.
+
+        The HTTP 200 response contains a single dict that contains keys that
+        are the course's advanced settings. For each setting a dictionary is
+        returned that contains the following fields:
+
+        * **deprecated**: This is true for settings that are deprecated.
+        * **display_name**: This is a friendly name for the setting.
+        * **help**: Contains help text that explains how the setting works.
+        * **value**: Contains the value of the setting. The exact format
+          depends on the setting and is often explained in the ``help`` field
+          above.
+
+        There may be other fields returned by the response.
+
+        **Example Response**
+
+        ```json
+        {
+            "display_name": {
```

Comment:
I am not aware if there was a thought process given to returned schema from this endpoint? has there been any discussion/decision made here? can you link me to that conversation so that I can follow it too?

Reply:
The aim was to leave the existing schema untouched.  It's somewhat discussed in the ADR for this PR: https://github.com/edx/edx-platform/pull/27129

This is the schema returned by the existing endpoint, and here we're just reusing the existing code.